DesignSafe-CI/portal
# task/DES-1925: add microsurvey popup for every 10 downloads in Publications

## Overview: ##

## PR Status: ##


## Related Jira tickets: ##

* [DES-1925](https://jira.tacc.utexas.edu/browse/DES-1925)

## Summary of Changes: ##
- Adds a survey that pops up on every 10th download in Data Files
- 2 new models: `DataFilesSurveyResult` (holds survey results) and `DataFilesSurveyCounter` (counter to tell when 10 downloads have occurred). 

## Testing Steps: ##
1.  Run migrations (`python manage.py migrate`)
2. In the Django shell instantiate the counter:
- ` from designsafe.apps.api.datafiles.models import DataFilesSurveyCounter`
- `c = DataFilesSurveyCounter(count=-1)`
- `c.save()`
3. In Publications, try downloading a file in 3 ways and make sure the survey popup appears:
- Toolbar button
- Preview modal button
- "Download Dataset" button
4. Try downloading again and make sure that the 2nd time doesn't pop up a survey.
5. Check that the survey saved in the database:
- ` from designsafe.apps.api.datafiles.models import DataFilesSurveyResult`
- `print(DataFilesSurveyResult.objects.all())` (should be a non-empty queryset)
6. To reset the counter and re-enable the popup:
- `c = DataFilesSurveyCounter.objects.all()[0]`
- `c.count = -1`
- `c.save()`
